### PR TITLE
provide some low level APIs to control RTC0 operation

### PR DESCRIPTION
**Pull request by liemleba**

### CODE REVIEW

File: include/hal/rtc.h [hunk 1]
```diff
@@ -3,32 +3,90 @@
...
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+#define __HAL_DISABLE_INTERRUPTS(x)                     \
+    do {                                                \
+        x = __get_PRIMASK();                            \
+        __disable_irq();                                \
+    } while(0);
```
Comment (vkbakken) [inline include/hal/rtc.h]: I do not want this code here. The rtc.h file is the common API for all devices defied in the RTC0 folder. These things are device specific.

File: include/hal/rtc.h [hunk 2]
```diff
@@ -3,32 +3,90 @@
...
-	uint32_t expiry;
-        TAILQ_ENTRY(hal_rtc_timer) link;    /* Queue linked list structure */
-};
+#define __HAL_ENABLE_INTERRUPTS(x)                      \
+    do {                                                \
+        if (!x) {                                       \
+            __enable_irq();                             \
+        }                                               \
+    } while(0);
+
```
Comment (vkbakken) [inline include/hal/rtc.h]: I do not want this code here. The rtc.h file is the common API for all devices defied in the RTC0 folder. These things are device specific.

File: include/hal/rtc.h [hunk 3]
```diff
@@ -3,32 +3,90 @@
...
+typedef enum
+{
+    /*lint -save -e30*/
+    NRF_RTC_EVENT_TICK        = offsetof(NRF_RTC_Type,EVENTS_TICK),       /**< Tick event. */
+    NRF_RTC_EVENT_OVERFLOW    = offsetof(NRF_RTC_Type,EVENTS_OVRFLW),     /**< Overflow event. */
+    NRF_RTC_EVENT_COMPARE_0   = offsetof(NRF_RTC_Type,EVENTS_COMPARE[0]), /**< Compare 0 event. */
+    NRF_RTC_EVENT_COMPARE_1   = offsetof(NRF_RTC_Type,EVENTS_COMPARE[1]), /**< Compare 1 event. */
+    NRF_RTC_EVENT_COMPARE_2   = offsetof(NRF_RTC_Type,EVENTS_COMPARE[2]), /**< Compare 2 event. */
+    NRF_RTC_EVENT_COMPARE_3   = offsetof(NRF_RTC_Type,EVENTS_COMPARE[3])  /**< Compare 3 event. */
+    /*lint -restore*/
```
Comment (vkbakken) [inline include/hal/rtc.h]: I do not want this code here. The rtc.h file is the common API for all devices defied in the RTC0 folder. These things are device specific.

File: rtc/hal_rtc_nrf5x.c [hunk 4]
```diff
@@ -140,150 +128,150 @@ void hal_rtc_init(void)
...
+//bool hal_rtc_start_at(struct hal_rtc_timer *timer, uint32_t tick)
+//{
+//	uint32_t ctx;
+//	struct hal_rtc_timer *entry;
+//    
+//	if ((timer == NULL) || (timer->link.tqe_prev != NULL) ||
+//			(timer->cb_fun == NULL)) {
+//		return false;
+//	}
+//
```
Comment (vkbakken) [inline rtc/hal_rtc_nrf5x.c:259]: Is all this code supposed to be commented away?

File: include/hal/rtc.h (hunk 1 above)
Comment (liemleba) [reply include/hal/rtc.h]: So, Would you mind if I move those Macros to freertos.c source file ?

File: rtc/hal_rtc_nrf5x.c (hunk 4 above)
Comment (liemleba) [reply rtc/hal_rtc_nrf5x.c:259]: No, It does not. I commented them away just to test if the device's firmware use it, and to show you that we are currently not using them. So, Do you want to restore those code ?

**Pull request closed without merging**